qiskit/providers/fake_provider/fake_generic.py

```diff
@@ -0,0 +1,525 @@
+        # corresponding to each calibrated instruction. Note that the calibration pulses
+        # are different for 1q gates vs 2q gates vs measurement instructions.
+        cmd_def = []
+        for inst in calibration_buffer:
+            num_qubits = self._supported_gates[inst].num_qubits
+            qarg_set = self._coupling_map if num_qubits > 1 else list(range(self.num_qubits))
+            if inst == "measure":
+                sequence = []
+                qubits = qarg_set
+                if inst in calibrate_instructions:
+                    sequence = [
+                        PulseQobjInstruction(
+                            name="acquire",
+                            duration=1792,
+                            t0=0,
+                            qubits=list(range(self.num_qubits)),
+                            memory_slot=list(range(self.num_qubits)),
+                        )
+                    ] + [PulseQobjInstruction(name="pulse_2", ch=f"m{i}", t0=0) for i in qarg_set]
+                cmd_def.append(
+                    Command(
+                        name=inst,
+                        qubits=qubits,
+                        sequence=sequence,
+                    )
+                )
+            else:
+                for qarg in qarg_set:
+                    sequence = []
+                    qubits = [qarg] if num_qubits == 1 else qarg
+                    if inst in calibrate_instructions:
+                        if num_qubits == 1:
+                            sequence = [
+                                PulseQobjInstruction(name="fc", ch=f"u{qarg}", t0=0, phase="-P0"),
+                                PulseQobjInstruction(name="pulse_1", ch=f"d{qarg}", t0=0),
+                            ]
+                        else:
+                            sequence = [
+                                PulseQobjInstruction(name="pulse_2", ch=f"d{qarg[0]}", t0=0),
+                                PulseQobjInstruction(name="pulse_3", ch=f"u{qarg[0]}", t0=0),
+                                PulseQobjInstruction(name="pulse_2", ch=f"d{qarg[1]}", t0=0),
+                                PulseQobjInstruction(name="fc", ch=f"d{qarg[1]}", t0=0, phase=2.1),
+                            ]
+                    cmd_def.append(
+                        Command(
+                            name=inst,
+                            qubits=qubits,
+                            sequence=sequence,
+                        )
+                    )
+
+        qubit_freq_est = np.random.normal(4.8, scale=0.01, size=self.num_qubits).tolist()
+        meas_freq_est = np.linspace(6.4, 6.6, self.num_qubits).tolist()
+        return PulseDefaults(
+            qubit_freq_est=qubit_freq_est,
+            meas_freq_est=meas_freq_est,
+            buffer=0,
+            pulse_library=pulse_library,
+            cmd_def=cmd_def,
+        )
+
+    def add_calibrations_to_target(self, inst_map: InstructionScheduleMap) -> None:
```

Comment:
The ability to provide instruction schedule maps to get calibrations from must be there somehow, most of our pulse tests were designed with a very specific set of calibrations in mind, or rely on making a v1 vs v2 comparison, see `test_macros.py`, for example: https://github.com/Qiskit/qiskit/pull/10266/files#diff-d9fcd472f8750bb1740ce3ca19151bc3724578854ebf03bb709bb3e10d03f659R44 (you can also see that this method is used a lot in #10918). I originally added an "`instruction_schedule_map`" input argument, but ended up scraping it because:

1. At the time some tests required more flexibility to alter individual calibrations.
2. I still needed a way to "turn off" calibrations and provide default calibrations for individual gates (that's why `calibrate_instructions` is a list, which I have never liked), but it looked awkward to have both `calibrate_instructions` and `instruction_schedule_map` in the signature. 

I ended up finding a way around altering individual calibrations by modifying the pulse tests, so all we really need now is a way to "turn off" default calibrations and a way to provide instruction schedule maps. I think that we could have `calibrate_instructions` be a bool instead of a list (some tests rely on the backend not having calibrations), or we could ever squeeze both options into one and allow `calibrate_instructions` to be either a bool or an `InstructionScheduleMap`.